### PR TITLE
feat: PLG Experiments 

### PR DESCRIPTION
**Pull request by devinder-harness**

### CODE REVIEW

File: src/pages/SignUp/SaasExperimentalForms/BasicLayout/BasicLayoutExperimental.tsx [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+  const moduleDetails = getModuleDetails(module as string);
+
+  return (
+    <div className={css.layoutExperimental}>
+      <div
+        className={css.imageColumn}
+        style={{ background: `url(${bgImg}) repeat` }}
+      >
+        {moduleDetails.logo && (
```
Comment (nanaxuHarness) [inline src/pages/SignUp/SaasExperimentalForms/BasicLayout/BasicLayoutExperimental.tsx:34]: seems the function will always return a logo, so we don't need the condition here

File: src/pages/SignUp/SaasExperimentalForms/SignUpExperimental.tsx [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+    edition?: string;
+    billingFrequency?: string;
+    utm_source?: string;
+    utm_content?: string;
+    utm_medium?: string;
+    utm_term?: string;
+    utm_campaign?: string;
+  }>();
+
+  const moduleDetails = getModuleDetails(module as string);
```
Comment (nanaxuHarness) [inline src/pages/SignUp/SaasExperimentalForms/SignUpExperimental.tsx:66]: same here

File: src/pages/SignUp/SaasExperimentalForms/SignUpExperimental.tsx [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+        utmInfo: {
+          utmSource: utm_source,
+          utmContent: utm_content,
+          utmMedium: utm_medium,
+          utmTerm: utm_term,
+          utmCampaign: utm_campaign
+        }
+      };
+
+      if (signupAction && signupAction.toUpperCase() in SignupAction) {
```
Comment (nanaxuHarness) [inline src/pages/SignUp/SaasExperimentalForms/SignUpExperimental.tsx:96]: can we use signupAction?.toUpperCase() here?

File: src/pages/SignUp/SaasExperimentalForms/SignUpExperimental.tsx [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+      data.email = data.email.toLowerCase();
+      setSignupData(data);
+      telemetry.track({
+        event: EVENT.SIGNUP_SUBMIT,
+        properties: {
+          intent: module || "",
+          category: CATEGORY.SIGNUP,
+          userId: data.email,
+          groupId: data.email,
+          utm_source: utm_source || "",
```
Comment (nanaxuHarness) [inline src/pages/SignUp/SaasExperimentalForms/SignUpExperimental.tsx:158]: we added a logic to read from local cookies for utm data, would you add it here as well

File: src/services/ng/overrides.yaml [hunk 5]
```diff
@@ -3,3 +3,4 @@ allowpaths:
   - "/signup/complete/{token}"
   - "/signup/verify-notification"
   - "/signup/community"
+  - "/users/feature-flags/{accountId}.get"
```
Comment (nanaxuHarness) [inline src/services/ng/overrides.yaml:6]: are we using this api in this PR?

**Comment by nanaxuHarness** [comment]
[src/static/images/image (3).svg] pls change this name to a meaningful one

File: src/pages/SignUp/SaasExperimentalForms/BasicLayout/BasicLayoutExperimental.tsx [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+  name?: string;
+  bgImg?: string;
+}
+const BasicLayoutExperimental: React.FC<BasicLayoutExperimentalProps> = ({
+  children,
+  bgImg = bgImageleft
+}) => {
+  const { module } = useQueryParams();
+
+  const moduleDetails = getModuleDetails(module as string);
```
[the remark this answers is not in the document]
Comment (devinder-harness) [reply src/pages/SignUp/SaasExperimentalForms/BasicLayout/BasicLayoutExperimental.tsx:26]: image
it is fine as we are falling back to `default`

File: src/services/ng/overrides.yaml (hunk 5 above)
Comment (devinder-harness) [reply src/services/ng/overrides.yaml:6]: updated

File: src/pages/SignUp/SaasExperimentalForms/SignUpExperimental.tsx (hunk 4 above)
Comment (devinder-harness) [reply src/pages/SignUp/SaasExperimentalForms/SignUpExperimental.tsx:158]: checking

File: src/pages/SignUp/SaasExperimentalForms/BasicLayout/BasicLayoutExperimental.tsx (hunk 6 above)
[the remark this answers is not in the document]
Comment (nanaxuHarness) [reply src/pages/SignUp/SaasExperimentalForms/BasicLayout/BasicLayoutExperimental.tsx:26]: hmm, I think the point here is not moduleDetails, it is `module as string` the type casting will throw exception if `const { module } = useQueryParams();`, module here is undefined.

File: src/pages/SignUp/SaasExperimentalForms/BasicLayout/BasicLayoutExperimental.tsx (hunk 1 above)
Comment (nanaxuHarness) [reply src/pages/SignUp/SaasExperimentalForms/BasicLayout/BasicLayoutExperimental.tsx:34]: `moduleDetails.logo &&` is redundant here, isn't it?

File: src/pages/SignUp/SaasExperimentalForms/SignUpExperimental.tsx (hunk 2 above)
Comment (nanaxuHarness) [reply src/pages/SignUp/SaasExperimentalForms/SignUpExperimental.tsx:77]: `module` here can be undefined

File: src/pages/SignUp/SaasExperimentalForms/SignUpExperimental.tsx [hunk 7]
```diff
@@ -0,0 +1,229 @@
...
+  function handleRecaptchaError() {
+    // Block the user until they refresh
+    setCaptchaExecuting(true);
+    handleError("Captcha has failed, please refresh the page.");
+  }
+
+  return (
+    <BasicLayoutExperimental>
+      <div className={cx(css.signup, css.experimentSignup)}>
+        {/* <div className={css.header}>
```
[the remark this answers is not in the document]
Comment (nanaxuHarness) [reply src/pages/SignUp/SaasExperimentalForms/SignUpExperimental.tsx:202]: im still seeing it here, did you push the change?

**Pull request merged**